v2/tracing.go

```diff
@@ -3,42 +3,76 @@ package shuttle
 import (
 	"context"
 
+	"go.opentelemetry.io/otel"
+
 	"github.com/Azure/azure-sdk-for-go/sdk/messaging/azservicebus"
-	"github.com/Azure/go-shuttle/v2/otel"
+	shuttleotel "github.com/Azure/go-shuttle/v2/otel"
 	"go.opentelemetry.io/otel/trace"
 )
 
+const (
+	serviceTracerName             = "go-shuttle"
+	defaultReceiverHandleSpanName = "receiver.Handle"
+)
+
 type TracingHandlerOpts struct {
-	enrichSpan func(context.Context, trace.Span)
+	spanStartOptions []trace.SpanStartOption
+	traceProvider    trace.TracerProvider
```

Comment:
allow to set the traceprovider. that's a better pattern.
if it's not set, we still grab the "global tracer"